src/database/DataTypes/Transaction.js

```diff
@@ -163,19 +163,17 @@ export class Transaction extends Realm.Object {
   }
 
   /**
-   * Delete any items that aren't contributing to this transaction, in order to
+   * Delete any batches and items that aren't contributing to this transaction, in order to
    * remove clutter
    * @param  {Realm} database   App wide local database
    * @return {none}
    */
-  pruneRedundantItems(database) {
-    const itemsToPrune = [];
-    this.items.forEach(transactionItem => {
-      if (transactionItem.totalQuantity === 0) {
-        itemsToPrune.push(transactionItem);
-      }
-    });
-    database.delete('TransactionItem', itemsToPrune);
+  pruneRedundantBatches(database) {
+    const batchesToRemove = this.getTransactionBatches(database)
+                              .filtered('numberOfPacks = 0');
+
+    database.delete('TransactionBatch', batchesToRemove);
```

Comment:
Exactly! We want as much of the clean up logic to be handle inside the destructors, that's why I designed the database to have destructors in the first place. Saves having to think about what you need to clean up every time, and then forgetting :-)